sympy/simplify/tests/test_simplify.py

```diff
@@ -1052,13 +1052,16 @@ def p(n,k):
     half = Float('0.5', 4)
     z = log(p(n, k) / p(n, k + 1)).expand(force=True)
     r = simplify(z.subs(n, N).n(4))
-    assert r == (
+    result = (
         half*k*log(k)
         - half*k*log(k + 1)
         + half*log(N)
         - half*log(k + 1)
         + Float(0.9189224, 4)
     )
+    tolerance = 1e-4
+
+    assert abs(r - result) < tolerance
```

Comment:
It is failing, but don't know the reason behind it.

Reply:
Hi, what message do you get? I've dealt with precision errors with `math.isclose()` or `numpy.allclose()` for some of my previous prs.

Reply:
This one https://github.com/sympy/sympy/pull/25442#issuecomment-1703977937

Reply:
Going through the comment maybe the pitfall is the difference between `python` float and `SymPy` float. Some examples
```
6.2  # Python float. Notice the floating point accuracy problems.
6.2000000000000002
S(6.2)  # SymPy Float has no such problems because of arbitrary precision.
6.20000000000000
```
or
```
>>> x = Symbol('x')
>>> x**(S(1)/2) == x**(0.5)
False
```

Still I'll go through the code and try to reproduce the failing test. Then get back to this !